src/lib.rs

```diff
@@ -205,25 +235,92 @@ pub async fn init_redis_connection(
     ip_addr: IpAddr,
     port: u16,
     client_id: String,
+    pool_options: Option<RedisConnPoolOptions>,
 ) -> anyhow::Result<()> {
     let manager = RedisConnectionManager::new(format!("redis://{ip_addr}:{port}"))?;
-    let pool = Pool::builder().build(manager).await?;
+    let pool = Pool::builder()
```

Comment:
let's do it like this.

```
let pool = build_redis_pool_with_options(); 
```

```
async fn build_pool_with_redis_conn_pool_options<M: ManageConnection>(
    manager: M,
    pool_options: Option<RedisConnPoolOptions<M>>,
) -> Result<Pool<M>> { .. }
```

Reply:
And inside `build_pool_with_redis_conn_pool_options`, we can bifurcate.
if `pool_options == None`, then we are going to use our own recommended default options to initialize `Pool`,
if `pool_options == Some`, then we are going to use the user's provided setting, while using our recommended default values for the options user did not specify, to initialize `Pool`.

Reply:
Got it! can do

Reply:
Done!